src/model/wallet.ts

```diff
@@ -414,11 +422,28 @@ const loadPrivateKey = async (): Promise<null | EthereumPrivateKey | -1> => {
       if (!address) {
         return null;
       }
+
       const privateKeyData = await getPrivateKey(address);
       if (privateKeyData === -1) {
         return -1;
       }
       privateKey = get(privateKeyData, 'privateKey', null);
+
+      let userPIN = null;
+      if (android) {
+        const hasBiometricsEnabled = await getSupportedBiometryType();
```

Comment:
Not urgent / we can do this in a separate PR, but some of these PIN checks are very similar to each other in multiple places in this file, and would be nice to have a separate function if it makes sense to do so

Reply:
I agree this can be improved but I don't wanna touch it now cause it has been QA'ed already and it's a lot of cases.
I'll keep this in mind if we introduce PIN for iOS or lockscreen.